code/modules/wiremod/components/delay.dm

```diff
@@ -0,0 +1,43 @@
+/**
+ * # Delay Component
+ *
+ * Delays a signal by a specified duration.
+ */
+/obj/item/circuit_component/delay
+	display_name = "Delay"
+
+	/// Amount to delay by
+	var/datum/port/input/delay_amount
+	/// Input signal to fire the delay
+	var/datum/port/input/trigger
+
+	/// The output of the signal
+	var/datum/port/output/output
+
+/obj/item/circuit_component/delay/Initialize()
+	. = ..()
+	delay_amount = add_input_port("Delay", PORT_TYPE_ANY, FALSE)
+	trigger = add_input_port("Trigger", PORT_TYPE_ANY)
+
+	output = add_output_port("Result", PORT_TYPE_NUMBER)
+
+/obj/item/circuit_component/delay/Destroy()
+	output = null
+	trigger = null
+	delay_amount = null
+	return ..()
+
+/obj/item/circuit_component/delay/input_received(datum/port/input/port)
+	. = ..()
+	if(.)
+		return
+
+	if(!COMPONENT_TRIGGERED_BY(trigger))
+		return
+
+	var/delay = delay_amount.input_value
+	if(delay > COMP_DELAY_MIN_VALUE)
+		// Convert delay into deciseconds
+		addtimer(CALLBACK(output, /datum/port/output.proc/set_output, trigger.input_value), delay*10)
```

Comment:
Short timers, especially ones that can be repeated like this one in a circuit, are generally bad. We have [timer subsystems](https://github.com/tgstation/tgstation/pull/59073), which might help here if we put all of circuits on one timer.

Reply:
This is the delay component. The timer can vary. I'll look into the timer subsystem